catroid/src/main/java/org/catrobat/catroid/embroidery/ShareEmbroideryFile.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.PointF;
+import android.net.Uri;
+import android.support.annotation.VisibleForTesting;
+import android.support.v4.content.FileProvider;
+import android.util.Log;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.Constants;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class ShareEmbroideryFile {
+	private static final String TAG = ShareEmbroideryFile.class.getSimpleName();
+
+	public ShareEmbroideryFile() {
+	}
+
+	public void shareEmbroideryPoints(Context context, ArrayList<PointF> stitchPoints) {
+		DSTFileGenerator dstFileGenerator = new DSTFileGenerator();
+		String filename = ProjectManager.getInstance().getCurrentProject().getName() + ".dst";
```

Comment:
This could be a problem because `Project.getName()` can give you a name that is invalid for a file, e.g. `project/1`. That is why we added `Project.getDirectory()` which encodes all problemeatid special chars when we want to create the project directory. You could use 
```
FileMetaDataExtractor.encodeSpecialCharsForFileSystem(ProjectManager.getInstance().getCurrentProject().getName()))
```
or **even better**:
change the method so that you can put the filename as a parameter. this way it is way more fexible and not coupled to the `ProjectManager`. Think about it, when you do it like you do you always have to be sure that a `project` is set in the ProjectManager just for the filename. You can see it in the test that this is not a good idea because of all the setup you need for the current project to be set. This is really bad practice because it firstly requres a `ProjectManager` and a current project and it forces the programmer to implicitly do things in the correct sequence (i.e. set current project, then write file).

If on the other hand you have to get the project name, or in this case file name, first it is clearer for anyone working on this that they have to provide the file name. They can still fetch it from the current project but it is no longer the `ShareEmbroideryFile`'s problem. 

By the way, please come up with a more descriptive name for this class. Maybe something like `EmbroideryFileExporter`. And the way I see it this does two things: it writes the file and when it's done it shows an intent chooser. So we might want to split that up. However, please implement the changes I suggested above first, in the meantime I'll think about this a litte more.